components/vl53l1x/__init__.py

```diff
@@ -0,0 +1,116 @@
+from typing import Dict, Any
+from esphome.components import i2c
+from esphome.core import CORE
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.const import (
+    CONF_FREQUENCY,
+    CONF_ID,
+    CONF_I2C,
+    CONF_I2C_ID,
+    CONF_INTERRUPT,
+    CONF_OFFSET,
+    CONF_PINS,
+)
+import esphome.pins as pins
+
+DEPENDENCIES = ["i2c"]
+AUTO_LOAD = ["i2c"]
+MULTI_CONF = False  # TODO enable when we support multiple addresses
+
+vl53l1x_ns = cg.esphome_ns.namespace("vl53l1x")
+VL53L1X = vl53l1x_ns.class_("VL53L1X", cg.Component)
+
+CONF_AUTO = "auto"
+CONF_CALIBRATION = "calibration"
+CONF_RANGING_MODE = "ranging"
+CONF_XSHUT = "xshut"
+CONF_XTALK = "xtalk"
+
+Ranging = vl53l1x_ns.namespace("Ranging")
+RANGING_MODES = {
+    CONF_AUTO: CONF_AUTO,
+    "shortest": Ranging.Shortest,
+    "short": Ranging.Short,
+    "medium": Ranging.Medium,
+    "long": Ranging.Long,
+    "longer": Ranging.Longer,
+    "longest": Ranging.Longest,
+}
+
+int16_t = cv.int_range(min=-32768, max=32768)  # signed
+
+
+def NullableSchema(*args, default: Any = None, **kwargs):
+    """
+    Same as Schema but will convert nulls to empty objects. Useful when all the schema keys are optional.
+    Allows YAML lines to be commented out leaving an "empty dict" which is mistakenly parsed as None.
+    """
+
+    def none_to_empty(value):
+        if value is None:
+            return {} if default is None else default
+        raise cv.Invalid("Expected none")
+
+    return cv.Any(cv.Schema(*args, **kwargs), none_to_empty)
+
+
+CONFIG_SCHEMA = cv.Schema(
+    {
+        cv.GenerateID(): cv.declare_id(VL53L1X),
+        cv.Optional(CONF_PINS, default={}): NullableSchema(
+            {
+                cv.Optional(CONF_XSHUT): pins.gpio_input_pin_schema,
+                cv.Optional(CONF_INTERRUPT): pins.gpio_output_pin_schema,
+            }
+        ),
+        cv.Optional(CONF_CALIBRATION, default={}): NullableSchema(
+            {
+                cv.Optional(CONF_RANGING_MODE, default=CONF_AUTO): cv.enum(
+                    RANGING_MODES
+                ),
+                cv.Optional(CONF_XTALK): cv.uint16_t,
+                cv.Optional(CONF_OFFSET): int16_t,
+            }
+        ),
+    }
+).extend(i2c.i2c_device_schema(0x52))
```

Comment:
```suggestion
).extend(i2c.i2c_device_schema(0x29))
```

When i added 0x52 i was using a knock off sensor. Genuine VL53L1X have 0x29 as default address

Reply:
You're thinking of VL53L0X.

https://github.com/esphome/esphome/blob/5c65f9f9ad8af021f0d73cb0eb12babd2b784c1a/esphome/components/vl53l0x/sensor.py#L61
https://github.com/rneurink/VL53L1X_ULD/blob/master/src/VL53L1X_ULD.h#L148

Reply:
Nope the pololu as well as the GY-53 uses 0x29 and those are the two preferred sensors. Black PCB sometimes uses 0x52.